### ISSUE TITLE
Incorrect warnings with `no-extra-semicolons` rule

### ISSUE THREAD

**Issue by ntwb** [stylelint/stylelint#1282]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug where the expected warning returned is unexpected 😏 

This is more of an issue with _semicolon_ rules used in tandem are confusing each other (I think)

> Which rule, if any, is this issue related to?

These 3 rules
• `no-extra-semicolons` 
• `block-closing-brace-newline-before`
• `declaration-block-semicolon-newline-after`

> What CSS is needed to reproduce this issue?

```css
a {
	color: #fff;
	display: block;;
}

b {
	color: #fff;;
	display: block;
}
```

> What stylelint configuration is needed to reproduce this issue?

```js
{
"rules": {
  "block-closing-brace-newline-before": "always",
  "declaration-block-semicolon-newline-after": "always",
  "no-extra-semicolons": true
  }
}

```

> Which version of stylelint are you using?

`6.3.3`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint test.css --config .teststylelintrc`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No

> What did you expect to happen?

I expected the following warnings were flagged:
```shell
src/wp-admin/css/edit.css
3:14	⚠  Unexpected extra semicolon (no-extra-semicolons) [stylelint]
7:14	⚠  Unexpected extra semicolon (no-extra-semicolons) [stylelint]

```

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
src/wp-admin/css/edit.css
2:18	⚠  Expected newline before "}" (block-closing-brace-newline-before) [stylelint]
7:14	⚠  Expected newline after ";" (declaration-block-semicolon-newline-after) [stylelint]
7:14	⚠  Unexpected extra semicolon (no-extra-semicolons) [stylelint]
```

**Comment by evilebottnawi** [comment]
@ntwb i get 
```shell
test.css
 3:18  ✖  Unexpected extra semicolon   no-extra-semicolons                      
 7:15  ✖  Expected newline after ";"   declaration-block-semicolon-newline-after
 7:15  ✖  Unexpected extra semicolon   no-extra-semicolons
```

**Comment by evilebottnawi** [comment]
@ntwb `declaration-block-semicolon-newline-after` false positive

**Comment by ntwb** [comment]
Yes, a false positive, I suspect similar occurrences when used with all 7 *semicolon* rules

**Comment by evilebottnawi** [comment]
@ntwb i correct this in `block-closing-brace-newline-before` https://github.com/stylelint/stylelint/blob/master/CHANGELOG.md#633. Now see how this may fix.

**Comment by ntwb** [comment]
Just double checked via `npm list --depth=0` I'm using stylelint `6.3.3` and I get:

```css
src/templates/default/css/bbpress.css
 3:17  ✖  Unexpected extra semicolon   no-extra-semicolons                      
 7:14  ✖  Expected newline after ";"   declaration-block-semicolon-newline-after
 7:14  ✖  Unexpected extra semicolon   no-extra-semicolons
```

Using:
```js
{
"rules": {
  "block-closing-brace-newline-before": "always",
  "declaration-block-semicolon-newline-after": "always",
  "no-extra-semicolons": true
  }
}
```

Via CLI `stylelint test.css --config .teststylelintrc`

**Comment by evilebottnawi** [comment]
@ntwb have found a solution in the near future will do PR :smile_cat:

**Comment by ntwb** [comment]
A variation, without the `block-closing-brace-newline-before` rule, the same warnings:

```shell
src/templates/default/css/bbpress.css
 3:17  ✖  Unexpected extra semicolon   no-extra-semicolons                      
 7:14  ✖  Expected newline after ";"   declaration-block-semicolon-newline-after
 7:14  ✖  Unexpected extra semicolon   no-extra-semicolons
```

Using:
```js
{
"rules": {
  "declaration-block-semicolon-newline-after": "always",
  "no-extra-semicolons": true
  }
}
```

Via CLI `stylelint test.css --config .teststylelintrc`

**Comment by davidtheclark** [comment]
I've been thinking about this and I've come to believe that warnings from `declaration-block-semicolon-newline-after` and `declaration-block-semicolon-space-after` are not actually bugs. The rules are doing exactly what they're supposed to, which is ensure that semicolons are followed by some whitespace. 

I don't think it really matters whether those semicolons are "extra" or not. I don't think we need to build in accommodations for erroneous syntax, especially since we already have `no-extra-semicolons` to more directly point out the problem.

Do you agree @ntwb and @evilebottnawi? If so, let's close this issue and the accompanying PR.

**Comment by ntwb** [comment]
Yes, I'm fine with that having re-read the original issue, closing :+1
